### PR TITLE
Refactor game queries to return only what page uses

### PR DESCRIPTION
**Pull request by crericha**

### CODE REVIEW

File: models/game.rb [hunk 1]
```diff
@@ -89,9 +85,19 @@ def self.home_games(user, **opts)
...
+    kwargs[:limit] = 1000
     fetch(user ? LOGGED_IN_QUERY : LOGGED_OUT_QUERY, **kwargs,).all.sort_by(&:id).reverse
   end
 
+  def self.profile_games(user)
+    kwargs = {}
+    kwargs[:user_id] = user.id if user
+    kwargs[:status] = %w[new active archived finished]
+    kwargs[:limit] = 100
+    fetch(USER_QUERY, **kwargs,).all.sort_by(&:id).reverse
```
Comment (tobymao) [inline models/game.rb:98]: why is it doing a sort_by reverse here, it can be done in sql, fix that here and line 90?

File: models/game.rb [hunk 2]
```diff
@@ -89,9 +85,19 @@ def self.home_games(user, **opts)
...
     kwargs[:user_id] = user.id if user
     kwargs[:title] = opts['title'] != '' ? opts['title'] : nil
+    kwargs[:status] = %w[new active]
+    kwargs[:limit] = 1000
     fetch(user ? LOGGED_IN_QUERY : LOGGED_OUT_QUERY, **kwargs,).all.sort_by(&:id).reverse
   end
 
+  def self.profile_games(user)
+    kwargs = {}
+    kwargs[:user_id] = user.id if user
```
Comment (tobymao) [inline models/game.rb:95]: why wouldn't there be a user?

Comment (tobymao) [reply models/game.rb:95]: fetch(user_query, user_id: user.id)... we don't need the kwargs building

File: models/game.rb (hunk 1 above)
Comment (crericha) [reply models/game.rb:98]: On line 90, its multiple queries combined together. Is it possible to sort those after the unions?

File: models/game.rb (hunk 2 above)
Comment (crericha) [reply models/game.rb:95]: copy and paste

Comment (crericha) [reply models/game.rb]: Fixed

File: models/game.rb (hunk 1 above)
Comment (crericha) [reply models/game.rb]: I don't think the sort order even matters because the page itself orders by updated_at. I just removed them.

**Pull request merged**